model/src/main/java/com/basistech/rosette/apimodel/SentimentOptions.java

```diff
@@ -16,62 +16,51 @@
 
 package com.basistech.rosette.apimodel;
 
+import java.util.Objects;
+
 /**
- * Sentiment options
+ * Sentiment options.
  */
 public final class SentimentOptions {
 
-    private Boolean explain;
+    private Integer explanationCount;
 
     /**
      * Create a set of sentiment analysis options with default values.
-     * Note that {@code null} is used to represent defaults.
      */
     public SentimentOptions() {
         //
     }
 
     /**
-     * constructor for {@code SentimentOptions}
-     * @param explain whether to return explanation strings for the sentiment results returned
-     */
-    public SentimentOptions(Boolean explain) {
-        this.explain = explain;
-    }
-
-    /**
-     * get whether to return explanation strings for the sentiment results returned
-     * @return whether to return explanation strings for the sentiment results returned
+     * @return the maximum number of explanation tokens to return.
```

Comment:
I actually thought that an explanation was necessarily a single token, but
I'll change.


On Wed, Feb 24, 2016 at 3:49 PM, Joe Ho <notifications@github.com> wrote:

> In
> model/src/main/java/com/basistech/rosette/apimodel/SentimentOptions.java
> <https://github.com/rosette-api/java/pull/13#discussion_r54003907>:
>
> >       */
> >      public SentimentOptions() {
> >          //
> >      }
> >
> >      /**
> > -     * constructor for {@code SentimentOptions}
> > -     * @param explain whether to return explanation strings for the sentiment results returned
> > -     */
> > -    public SentimentOptions(Boolean explain) {
> > -        this.explain = explain;
> > -    }
> > -
> > -    /**
> > -     * get whether to return explanation strings for the sentiment results returned
> > -     * @return whether to return explanation strings for the sentiment results returned
> > +     * @return the maximum number of explanation tokens to return.
>
> I got confused with the word "tokens". I thought you are limiting the
> maximum words (tokens) in an explanation. In reality you are limiting the
> maximum explanations. Suggest to change to
>
> @return <https://github.com/return> the maximum number of explanations to
> return.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rosette-api/java/pull/13/files#r54003907>.
>